### ISSUE TITLE
Place named-screenshots under the relevant test screenshots folder rather than the screenshots root

### ISSUE THREAD

**Issue by adrianaisemberg** [DevExpress/testcafe#1974]
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
Taking named screenshots saves them under the screenshots root.
Compared to taking unnamed screenshots which saves those under the currently running test folder.

### What is the expected behavior?
Save named screenshots to the same place as the unnamed (indexed) ones, or at least under a sibling folder.

### How would you reproduce the current behavior (if this is a bug)?

1. Write two different and unrelated tests.
2. In each test, call to `t.takeScreenshot("after-click")` 
3. Run the tests
4. Observe the screenshots folder. Notice that only one file exists although we took two screenshots.
This is a result of the second test screenshot overwriting the first's.

This raises several concerns:
1. If we want to report the test results and include all relevant screenshots of a certain test - we don't have a way to collect test-specific relevant named screenshots from the results folder.
2. Using the same name results in overwriting screenshots. The only way to prevent this is to have strict naming constraints. However, those can't be enforced.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js

```

### Specify your

* operating system:
* testcafe version:
* node.js version:

**Comment by AndreyBelym** [comment]
Hi @adrianaisemberg, we are working out a way to enable more flexible screenshots path, but now you can use the following approach as a workaround:
```js
import { ClientFunction } from 'testcafe';
import { parse } from 'useragent';

fixture `Take a screenshot`
    .page `example.com`;

const getUserAgent = ClientFunction(() => navigator.userAgent.toString());

test('Take a screenshot with a custom path', async t => {
    const ua       = await getUserAgent();
    const parsedUA = parse(ua);

    await t.takeScreenshot('custom/' + parsedUA.family + '.png');
});
```

It will save screenshots to files according to browser names, e.g. `custom/Chrome.png`, `custom/Firefox.png`, etc.

**Comment by AlexanderMoskovkin** [comment]
closed in favor of #2152
